### PR TITLE
Add scheduleable yank function and tests

### PR DESCRIPTION
**Pull request by brianmcmichael**
Allow governance to schedule a yank at a future date.

### CODE REVIEW

File: src/DssVest.sol [hunk 1]
```diff
@@ -184,21 +184,35 @@ contract DssVest {
...
         require(wards[msg.sender] == 1 || awards[_id].mgr == msg.sender, "DssVest/not-authorized");
         Award memory _award = awards[_id];
         require(_award.usr != address(0), "DssVest/invalid-award");
-        awards[_id].fin = uint48(block.timestamp);
+        if (_end < block.timestamp) {
+            _end = block.timestamp;
+        } else if (_end > _award.fin) {
+            _end = _award.fin;
+        }
+        awards[_id].fin = uint48(_end);
```
Comment (hexonaut) [inline src/DssVest.sol:213]: Should be `toUint48(_end)`.

**Review by hexonaut — APPROVED** [review]

File: src/DssVest.sol [hunk 2]
```diff
@@ -184,21 +184,35 @@ contract DssVest {
...
+    function yank(uint256 _id, uint256 _end) public {
         require(wards[msg.sender] == 1 || awards[_id].mgr == msg.sender, "DssVest/not-authorized");
         Award memory _award = awards[_id];
         require(_award.usr != address(0), "DssVest/invalid-award");
-        awards[_id].fin = uint48(block.timestamp);
+        if (_end < block.timestamp) {
+            _end = block.timestamp;
+        } else if (_end > _award.fin) {
+            _end = _award.fin;
+        }
```
[the remark this answers is not in the document]
Comment (gbalabasquer) [reply src/DssVest.sol:212]: I mean adding it as a require in the beginning.

File: src/DssVest.t.sol [hunk 3]
```diff
@@ -307,6 +307,85 @@ contract DssVestTest is DSTest {
         assertEq(Token(address(vest.gem())).balanceOf(address(this)), 4 * days_vest);
     }
 
+    function testYankSchedulePassed() public {
+        uint256 id = vest.init(address(this), 100 * 10**18, block.timestamp, 100 days, 20 days, address(0));
+        uint256 days_vest = 1000000000000000000;
```
Comment (gbalabasquer) [inline src/DssVest.t.sol:312]: What's the reason of this variable, why not just use 10 ** 18 everywhere?

[the remark this answers is not in the document]
Comment (brianmcmichael) [reply src/DssVest.t.sol]: see aa825df485234f14ddfa22ec252302f39ce6aeeb

File: src/DssVest.t.sol [hunk 4]
```diff
@@ -282,8 +277,7 @@ contract DssVestTest is DSTest {
 
     function testYankAfterVest() public {
         // Test case where vest is yanked twice, say by manager and then governance
```
Comment (gbalabasquer) [inline src/DssVest.t.sol:279]: I don't think this comment is right, not seeing two `yank`s.

File: src/DssVest.sol [hunk 5]
```diff
@@ -184,21 +184,35 @@ contract DssVest {
...
+    */
+    function yank(uint256 _id, uint256 _end) public {
         require(wards[msg.sender] == 1 || awards[_id].mgr == msg.sender, "DssVest/not-authorized");
         Award memory _award = awards[_id];
         require(_award.usr != address(0), "DssVest/invalid-award");
-        awards[_id].fin = uint48(block.timestamp);
+        if (_end < block.timestamp) {
+            _end = block.timestamp;
+        } else if (_end > _award.fin) {
+            _end = _award.fin;
```
[the remark this answers is not in the document]
Comment (kmbarry1) [reply src/DssVest.sol:211]: Oh nevermind...I see we're using this to set `fin`.

**Review by gbalabasquer — APPROVED** [review]
LGTM. Just a minor comment about a comment that is not actually new from this PR.

**Pull request merged**